### DISCUSSION TITLE
`CUSTOM` in `ALTER TYPE`

### DISCUSSION THREAD

**Discussion by gramian** [ArcadeData/arcadedb#588] [Q&A] [answered]
Dear Community,

The [`ALTER TYPE`](https://docs.arcadedb.com/#SQL-Alter-Type) SQL statement has a `CUSTOM key = value` component, which I don't understand: There seems to be no alteration on the to-be-altered type if used. I assumed `CUSTOM` works similar to the SQL [`COMMENT ON`](http://aturing.umcs.maine.edu/~sudarshan.chawathe/200909/cos480/pgsql/doc/html/sql-comment.html) statement. Could you resolve what I am misunderstanding or misinterpreting here? Thank you.

PS: If it is supposed to work like `COMMENT ON`, is there a way use the equivalent of `COMMENT ON DATABASE IS`? I tried `ALTER TYPE schema:database CUSTOM comment = "a database"` which does not work; I assume `schema:database` is protected in some way.

**Comment by lvca** [comment, marked as answer]
CUSTOM are custom properties you can store within your type. It could be anything, comments, descriptions, or some properties you need in your application. CUSTOM properties are available on types and properties, not for the database.

**Reply by gramian** [reply]
Thank you for explaining,
So as I understand it (now), custom properties attach data (strings or integers) to types in a chosen key.

I made a type and altered a custom property with value; however; nothing changed when I checked the type:
```
CREATE DOCUMENT TYPE test;
ALTER TYPE test CUSTOM comment = "hi";
SELECT comment FROM schema:types;
```
I get nothing returned (in the studio). I was expecting to get back `"hi"`. Maybe this illustrates my problem of understanding better. I guess I am looking in the wrong place?

Thanks again

**Reply by gramian** [reply]
@lvca May be this is a bug?

**Reply by lvca** [reply]
It could be that the query against `schema:types` is not returning custom information. I'm looking at it.

**Reply by lvca** [reply]
That was the case. The custom properties are saved, but not returned from the schema:types. Opened this issue: https://github.com/ArcadeData/arcadedb/issues/590 and fixed right away.

**Comment by gramian** [comment]
> CUSTOM properties are available on types and properties

This would mean the component `[CUSTOM <custom-key> <custom-value>]`needs to be added to the [`ALTER PROPERTY`](https://docs.arcadedb.com/#SQL-Alter-Property) documentation?

**Reply by lvca** [reply]
Thanks, it was missing. Just added. Also in the grammar is not `[CUSTOM <custom-key> <custom-value>]` but `[CUSTOM <custom-key> =  <custom-value>]`
